src/BenchmarkDotNet/EventHandlers/EventHandlerBase.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using BenchmarkDotNet.Reports;
+using BenchmarkDotNet.Running;
+using BenchmarkDotNet.Toolchains.Results;
+using BenchmarkDotNet.Validators;
+
+namespace BenchmarkDotNet.EventHandlers
+{
+    public class EventHandlerBase : IEventHandler
```

Comment:
I know that all existing BDN config switches expose interfaces, but in this case I wonder whether we should just expose an `abstract` class with all the methods being `virtual`. Benefits:
- to sign up for a single event, the users would need to override only one method
- if we wanted to introduce a new event in the future, we would not break the existing users, as their code would still compile.

What do you think?

Reply:
I was about to say something very similar.  The fact that this is "one type to handle all kinds of events" means an interface is not suitable.